sdk/nodejs/output.ts

```diff
@@ -312,6 +236,77 @@ To manipulate the value of this Output, use '.apply' instead.`);
             },
         });
     }
+
+    public get(): T {
+        throw new Error(`Cannot call '.get' during update or preview.
+To manipulate the value of this Output, use '.apply' instead.`);
+    }
+
+    public apply<U>(func: (t: T) => Input<U>): Output<U> {
+        const applied: Promise<OutputData<U>> =
+            Promise.all([this.promise(), this.isKnown, this.isSecret])
+                   .then(([value, isKnown, isSecret]) => applyHelperAsync<T, U>(value, isKnown, isSecret, func));
+
+        const result = new OutputImpl<U>(
+            this.resources(),
+            applied.then(a => a.value),
+            applied.then(a => a.isKnown),
+            applied.then(a => a.isSecret));
```

Comment:
we just grab out the data from the promise we get back and make the new Output out of it

Note: with this PR we can see how easy it would be to make 'resources' into a Promise as well in case we want the inner func to be able to affect it.